bin/sign.py

```diff
@@ -226,6 +229,36 @@ def json_tempfile(path: Path, data: JSON) -> Generator[None, None, None]:
     path.unlink()
 
 
+def parse_env_vars(runtime: str) -> tuple[str, str, str, JSON]:
+    """Parse and return environment variables"""
+    try:
+        aad_id = os.environ["SIGNING_AAD_ID"]
+        tenant_id = os.environ["SIGNING_TENANT_ID"]
+        customer_correlation_id = os.environ["SIGNING_CUSTOMER_CORRELATION_ID"]
+        match runtime:
+            case "win10-x64":
+                # This key code is used for signing .exes and .dlls on both Windows and Mac.
+                key_codes = {

```

Comment:
In Python, does defining `key_codes` inside the match arms like this also define `key_codes` in the surround lexical scope? @mvanchaa @reillysiemens

Reply:
I'm not sure if I understood what you meant, I tried this now and the `key_codes` is not defined outside of match block if it's not executed in any of the case statements. Is that what you meant?

Reply:
@kyle-rader-msft: Yes.
```python
>>> x
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'x' is not defined
>>> match False:
...     case True:
...         x = 42
...     case False:
...         pass
...
>>> x
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'x' is not defined
>>> match True:
...     case True:
...         x = 42
...     case False:
...         pass
...
>>> x
42
```